be/src/storage/tablet_manager.cpp

```diff
@@ -983,28 +983,35 @@ Status TabletManager::report_all_tablets_info(std::map<TTabletId, TTablet>* tabl
 
     StarRocksMetrics::instance()->report_all_tablets_requests_total.increment(1);
 
-    size_t max_tablet_rowset_num = 0;
+    std::vector<TabletSharedPtr> all_tablets;
```

Comment:
may cause this vector size too large. move this to tablet_shard level. so the under the lock, just get the array from the tablet_shard, and then generate the report info without lock.

This way the vector size will be the max(tablets_shards) instead of sum(tablets_shards);

Reply:
updated